File: front-end/src/shared/types/UpdateQuestionDTO.ts
import type { QuestionType } from "@/shared/types/QuestionType";
import type { QuestionConfigDTO } from "@/shared/types/QuestionConfigDTO";
import type { CreateQuestionOptionDTO } from "@/shared/types/CreateQuestionOptionDTO";

export interface UpdateQuestionDTO {
  id?: number;
  type: QuestionType;
  questionText?: string;
  explanation?: string;
  imageUrl?: string;
  videoUrl?: string;
  points?: number;
  order?: number;
  isRequired?: boolean;
  config?: QuestionConfigDTO;
  options?: CreateQuestionOptionDTO[];
}

File: back-end/src/main/java/com/meutcc/backend/content/question/dtos/UpdateQuestionDTO.java
package com.meutcc.backend.content.question.dtos;

import com.meutcc.backend.content.question.QuestionType;
import jakarta.validation.constraints.Min;

import java.util.List;

public record UpdateQuestionDTO(
        QuestionType type,
        String questionText,
        String explanation,
        String imageUrl,
        String videoUrl,

        @Min(value = 1, message = "Pontos devem ser maior que 0")
        Integer points,

        @Min(value = 0, message = "Ordem deve ser maior ou igual a 0")
        Integer order,

        Boolean isRequired,
        QuestionConfigDTO config,
        List<CreateQuestionOptionDTO> options
) {
}
